### ISSUE TITLE
Not possible to open the settings

### ISSUE THREAD

**Issue by sam9032** [vchlum/hue-lights#32]
Hey there,

it isn't possible for me to open the extension settings:
![Screenshot from 2021-09-27 19-09-44](https://user-images.githubusercontent.com/12381166/134954622-f7e0eb04-c1ae-4cef-8509-dc86fd15a65d.png)

The displayed Technical Details:

The settings of extension hue-lights@chlumskyvaclav.gmail.com had an error:
```
TypeError: session.send_message is not a function
```

Stack trace:
```
discoverBridges@/home/sam/.local/share/gnome-shell/extensions/hue-lights@chlumskyvaclav.gmail.com/phueapi.js:58:30
checkBridges@/home/sam/.local/share/gnome-shell/extensions/hue-lights@chlumskyvaclav.gmail.com/phue.js:209:33
HuePrefs@/home/sam/.local/share/gnome-shell/extensions/hue-lights@chlumskyvaclav.gmail.com/prefs.js:126:19
buildPrefsWidget@/home/sam/.local/share/gnome-shell/extensions/hue-lights@chlumskyvaclav.gmail.com/prefs.js:2021:20
_init@resource:///org/gnome/Shell/Extensions/js/extensionsService.js:209:40
OpenExtensionPrefsAsync/<@resource:///org/gnome/Shell/Extensions/js/extensionsService.js:122:28
asyncCallback@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:115:22
run@resource:///org/gnome/Shell/Extensions/js/dbusService.js:177:20
main@resource:///org/gnome/Shell/Extensions/js/main.js:19:13
run@resource:///org/gnome/gjs/modules/script/package.js:206:19
start@resource:///org/gnome/gjs/modules/script/package.js:190:8
@/usr/share/gnome-shell/org.gnome.Shell.Extensions:1:17
```

I use Arch with Gnome 40.4.

**Comment by vchlum** [comment]
Hi @sam9032! Thank you for reporting the issue! I can confirm the issue. As I can see, Arch Linux has just updated the libsoup2.4 to the new libsoup3. It brings major changes and backward incompatibilities. I am going to fix it as soon as possible (depends on my free time). Unfortunately, It requires a significant amount of work.

**Comment by vchlum** [comment]
@sam9032 I have a workaround for you or others with the same issue... You can install the package libsoup 2 like this:

`sudo pacman -S libsoup`

and run following command:

`sed -i '/const Soup = imports.gi.Soup;/i imports.gi.versions.Soup = "2.4";' .local/share/gnome-shell/extensions/hue-lights\@chlumskyvaclav.gmail.com/*.js`

**Comment by sam9032** [comment]
Hi @vchlum,
thank you! I'll try that tomorrow, but had the same thought, to force the use of the older libsoup version.
Your extension looks really cool, so that would be amazing!
Thank you for your time.

**Comment by sam9032** [comment]
Your workaround works fine for me! Libsoup 2.74 which provides libsoup-2.4 was already installed so I only need to change the line of code in your extension.

**Comment by vchlum** [comment]
I did some research. libsoup 3 seems to be very rare in Linux distributions yet. Seems like [Arch and ALT provide it](https://pkgs.org/search/?q=libsoup3) only (today), and the libsoup 2 is not removed yet. I added the preference to use version 2.4 and partial implementation of libsoup 3.

Seems like libsoup 3 can be used only in the settings, which works with the last patch. In the extension menu, I wasn't able to use and debug libsoup 3 even with `imports.gi.versions.Soup = "3.0"`. It uses version 2 anyway. (on arch linux)

The libsoup 3 proper support will come once needed. For now, this is resolved. I will keep an eye on libsoup 3 spreading. Thanks again for reporting.
